File: TOKI-Learning/c++/gaya_penulisan_variabel.cpp
#include <cstdio>
#include <string>

using namespace std;

void toCamelCase(string &s)
{
    int underscore = s.find("_");
    while (underscore != -1)
    {
        s[underscore + 1] -= 'a' - 'A';
        s.erase(underscore, 1);
        underscore = s.find("_");
    }
}

void to_snake_case(string &s)
{
    for (int i = 0; i < s.length(); i++)
        if (s[i] <= 'Z')
        {
            s[i] -= 'A' - 'a';
            s.insert(i, "_");
        }
}

int main()
{
    char buff[101];
    scanf("%s", buff);
    string s = buff;
    if (s.find("_") != -1)
        toCamelCase(s);
    else
        to_snake_case(s);
    printf("%s\n", s.c_str());
    return 0;
}

File: TOKI-Learning/pascal/gaya_penulisan_variabel.pas
var
	i : integer;
	s : string;
	
function toCamelCase(a :string):string;
begin
	i := 1;
	while i <= length(a) do
	begin
		if a[i] = '_' then
		begin
			delete(a,i,1);
			a[i] := upcase(a[i]);
		end;
		inc(i);
	end;
	toCamelCase := a;
end;

function to_snake_case(a: string):string;
begin
	i := 1;
	while i <= length(a) do
	begin
		if ord(a[i]) <= ord('Z') then
		begin
			insert('_',a,i);
			a[i+1] := lowercase(a[i+1]);
		end;
		inc(i);
	end;
	to_snake_case := a;
end;

begin
	readln(s);
	if pos('_',s) <> 0 then
		writeln(toCamelCase(s))
	else
		writeln(to_snake_case(s));
end.